src/rnnlm/rnnlm-utils-test.cc

```diff
@@ -0,0 +1,177 @@
+// rnnlm/rnnlm-utils-test.cc
+
+#include <math.h>
+#include "rnnlm/rnnlm-utils.h"
+
+namespace kaldi {
+namespace rnnlm {
+
+void PrepareVector(int n, int ones_size, std::set<int> &outputs_set, vector<BaseFloat> &selection_probs) {
```

Comment:
The kaldi style requires (in most cases) never passing references; 
either pass by const reference or pass by pointer if you want to change anything